optimum/habana/transformers/models/mistral/modeling_mistral.py

```diff
@@ -322,10 +387,16 @@ def forward(
 class GaudiMistralDecoderLayer(MistralDecoderLayer):
     def __init__(self, config: MistralConfig, layer_idx: int):
         super().__init__(config, layer_idx)
+        self.hidden_size = config.hidden_size
+
         self.self_attn = GaudiMistralAttention(config, layer_idx)
```

Comment:
With Llama, there was a memory issue here with DeepSpeed ZeRO-3. Can we do the same here and call super on `MistralDecoderLayer`? https://github.com/huggingface/optimum-habana/blob/051e5e5daada64393bcf17b59e69fc7178047542/optimum/habana/transformers/models/llama/modeling_llama.py#L476